trino/sqlalchemy/datatype.py

```diff
@@ -0,0 +1,169 @@
+
+        if isinstance(value_type, type):
+            value_type = value_type()
+        self.value_type: TypeEngine = value_type
+
+    @property
+    def python_type(self):
+        return dict
+
+
+class ROW(TypeEngine):
+    __visit_name__ = "ROW"
+
+    def __init__(self, attr_types: Dict[str, SQLType]):
+        for name, attr_type in attr_types.items():
+            if isinstance(attr_type, type):
+                attr_type = attr_type()
+                attr_types[name] = attr_type
+        self.attr_types: Dict[str, TypeEngine] = attr_types
+
+    @property
+    def python_type(self):
+        return dict
+
+
+def split(string: str, delimiter: str = ',',
+          quote: str = '"', escaped_quote: str = r'\"',
+          open_bracket: str = '(', close_bracket: str = ')') -> Iterator[str]:
+    """
+    A split function that is aware of quotes and brackets/parentheses.
+
+    :param string: string to split
+    :param delimiter: string defining where to split, usually a comma or space
+    :param quote: string, either a single or a double quote
+    :param escaped_quote: string representing an escaped quote
+    :param open_bracket: string, either [, {, < or (
+    :param close_bracket: string, either ], }, > or )
+    """
+    parens = 0
+    quotes = False
+    i = 0
+    for j, character in enumerate(string):
+        complete = parens == 0 and not quotes
+        if complete and character == delimiter:
+            yield string[i:j]
+            i = j + len(delimiter)
+        elif character == open_bracket:
+            parens += 1
+        elif character == close_bracket:
+            parens -= 1
+        elif character == quote:
+            if quotes and string[j - len(escaped_quote) + 1: j + 1] != escaped_quote:
+                quotes = False
+            elif not quotes:
+                quotes = True
+    yield string[i:]
+
+
+def parse_sqltype(type_str: str) -> TypeEngine:
+    type_str = type_str.strip().lower()
+    match = re.match(r'^(?P<type>\w+)\s*(?:\((?P<options>.*)\))?', type_str)
+    if not match:
+        util.warn(f"Could not parse type name '{type_str}'")
+        return sqltypes.NULLTYPE
+    type_name = match.group("type")
+    type_opts = match.group("options")
+
+    if type_name == "array":
+        item_type = parse_sqltype(type_opts)
+        if isinstance(item_type, sqltypes.ARRAY):
+            dimensions = (item_type.dimensions or 1) + 1
+            return sqltypes.ARRAY(item_type.item_type, dimensions=dimensions)
+        return sqltypes.ARRAY(item_type)
+    elif type_name == "map":
+        key_type_str, value_type_str = split(type_opts)
+        key_type = parse_sqltype(key_type_str)
+        value_type = parse_sqltype(value_type_str)
+        return MAP(key_type, value_type)
+    elif type_name == "row":
+        attr_types: Dict[str, SQLType] = {}
+        for attr_str in split(type_opts):
+            name, attr_type_str = split(attr_str.strip(), delimiter=' ')
```

Comment:
What about `ROW("a space" VARCHAR, "another space" INT)`? splitting on spaces seems incorrect?
Note that types can also be implicit as in `MAP(ARRAY['foo', 'bar'], ARRAY[1, 2])`.

My examples may not be valid though because I haven't yet understood who calls this class/method and with what kind of inputs.